### ISSUE TITLE
Output of autoregressive prior

### ISSUE THREAD

**Issue by harrygcoppock** [rosinality/vq-vae-2-pytorch#39]
First of all, thank you for the Implementation.

In your implementation, the output of the PixelSnail is a softmax layer which corresponds to the index of the codebook. I was wondering where the authors detail this? I cannot seem to find any reference to this. 

Thanks in advance.

**Comment by rosinality** [comment]
Original PixelSNAIL used mixture of discretized logistic loss. But as VQ-VAE prior works on arbitrary codes not pixel values, I though softmax is more appropriate.

**Comment by harrygcoppock** [comment]
Thank you for your speedy response. So you couldn't find any mention of the actual output of the PixelSnail the authors used?

**Comment by rosinality** [comment]
Yes, I don't know the exact settings authors have used.

**Comment by harrygcoppock** [comment]
I have just been in contact with the author of the paper (Oriol Vinyals) and can confirm that softmax was the correct implementation.  Thanks again for your help @rosinality

**Comment by rosinality** [comment]
Thank you. I also thought about possibilities that authors have used some special loss like label smoothing.

**Comment by harrygcoppock** [comment]
Yes as otherwise (from my experience) the model can be quite brittle, for example changing just a few of the discrete latents can completely disfigure the output. I was curious as I thought that maybe the output of the autoregressive prior would be: LatentHeightxLatentWidthxCodeBookVectorDepth(64) and then Vector Quantization would be applied to it.
